spec/features/omniauth_spec.rb

```diff
@@ -118,9 +118,43 @@
     end
   end
 
+  shared_examples 'sign up with openid provider' do |provider|
+    scenario "#{provider} has option to log in" do
+      visit '/accounts/sign_in'
+      expect(page.has_content?('or sign in using')).to eq true
+      expect(page.has_link?("omniauth-#{provider}")).to eq true
+    end
+
+    scenario "#{provider} signs up new user" do
+      expected_count_openid = Openid.count + 1
+      expected_count_user = User.count + 1
+      visit '/accounts/sign_in'
+
+      mock_auth_accounts
+      within('#openidlinks') do
+        click_link "omniauth-#{provider}"
+      end
+      expect(flash).to eq("user-#{provider}@example.com signed in successfully with #{provider}")
+      expect(Openid.count).to eq(expected_count_openid)
+      expect(User.count).to eq(expected_count_user)
+    end
+  end
+
   describe 'omniauth' do
     if User.omniauth_providers.present?
       it_behaves_like 'sign in with openid'
     end
   end
+
+  describe 'omniauth provider' do
+    providers = %w(facebook google suse github)
+
+    it 'verifies presence' do
```

Comment:
If it fails then someone will, naturally, edit the 'providers' variable to make it pass.
It is designed to cause the 'sign up with openid provider' test to fail if the contributor doesn't make the appropriate changes to 'omniauth_macros.rb' too.

I must leave a comment to remind future contributors to also edit 'development.rb'.

Shall I move it to a model test and count on future contributors and reviewers to make and check that the right changes have been made to the rest of the files, instead of enforcing it via technical means? (at least for 'omniauth_macros.rb')